### PR TITLE
Tsify sendFeedback and cleanup errors/notifications

### PR DESCRIPTION
**Pull request by nf-s**
### Tsify sendFeedback and cleanup errors/notifications

After vs before

image

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [ ] I've updated relevant documentation in `doc/`.
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/Models/sendFeedback.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        method: "POST",
+        data: JSON.stringify(feedbackData),
+        headers: {
+          "Content-Type": "application/json"
+        }
+      });
+    })
+    .then(function(json) {
+      if (json instanceof String) {
+        json = JSON.parse(json.toString());
```
Comment (wibily) [inline lib/Models/sendFeedback.ts:70]: this is subtly different to the original jsx version, although from a cursory test in the console `JSON.parse(new String('{"a":1}')` seems to work?

File: lib/Models/sendFeedback.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  email: string;
+  sendShareURL: boolean;
+  comment: string;
+  additionalParameters?: Record<string, string | undefined>;
+}) {
+  if (!isDefined(options) || !isDefined(options.terria)) {
+    throw TerriaError.from("options.terria is required.");
+  }
+
+  var terria = options.terria;
```
Comment (wibily) [inline lib/Models/sendFeedback.ts]: could be a const, but its such a nitpick, i'm not fussed

**Review by wibily — APPROVED** [review]

File: lib/Models/sendFeedback.ts (hunk 1 above)
Comment (nf-s) [reply lib/Models/sendFeedback.ts:70]: Yeah I was just trying to please the type checker here. Probably not necessary though.
I'm not sure how we'd even get a `String` here

Comment (nf-s) [reply lib/Models/sendFeedback.ts:70]: I might leave it, just to be careful

File: lib/Models/sendFeedback.ts (hunk 2 above)
Comment (nf-s) [reply lib/Models/sendFeedback.ts]: 👍

**Pull request merged**